kymatio/scattering1d/core/timefrequency_scattering.py

```diff
@@ -139,14 +143,18 @@ def timefrequency_scattering(
         k2 = max(j2 - k1 - oversampling, 0)  # what we subsample now in 2nd
         Y_2_c = S_1_c_list[n1]               # reuse 1st-order U_1_hat * phi[k1]
         Y_2_hat = B.subsample_fourier(Y_2_c, 2**k2)
-        Y_2_list.append(B.ifft(Y_2_hat))
+        Y_2_c = B.ifft(Y_2_hat)
+
+        if Y_2_arr is None:
+            Y_2_arr = backend.zeros((2**sc_freq.J_pad[-1], Y_2_c.shape[-1]),
```

Comment:
why initialize `Y_2_arr` with `None` instead of directly this `backend.zeros` array? that would spare an `if` statement in the for loop. Am i missing something important here?

Reply:
See [4 here](https://github.com/kymatio/kymatio/pull/674#issuecomment-815896118)

Reply:
ah, i see, you need `Y_2_c.shape[-1]`. ok!

Reply:
maybe remove the `# TODO` in L135 then?

Reply:
It's reminder to take another look if this can be computed easily, may drop with next commit